Editorial review: Document interest invokers (#42021)
mdn/content: files/en-us/web/css/reference/properties/interest-delay/index.md

/* Two values */
interest-delay: 1s normal;
interest-delay: 0s 500ms;

/* Global values */
interest-delay: inherit;
interest-delay: initial;
interest-delay: revert;
interest-delay: revert-layer;
interest-delay: unset;
```

### Values

The `interest-delay` property is specified using one or two values, which can be the keyword `normal` or a {{cssxref("&lt;time>")}} value:

- `normal`
  - : The initial value; specifies that the delays should be set to the browser's default value.
- {{cssxref("&lt;time>")}}
  - : Sets the delays to a specific duration.

## Description

A control element such as an {{htmlelement("a")}} or {{htmlelement("button")}} can be specified as an [interest invoker](/en-US/docs/Web/API/Popover_API/Using_interest_invokers) by giving it an [`interestfor`](/en-US/docs/Web/HTML/Reference/Elements/a#interestfor) attribute with a value equal to the `id` of a target element. When this relationship is established, the target element will be affected in some way when the user "shows interest" in the invoker (for example by hovering or focusing it). A common use case is to cause a popover to appear on hover/focus. When the user "loses interest", the effect will stop.

> When the user shows/loses interest, the associated effect doesn't start/stop immediately — the browser introduces a short delay, which may vary by vendor. This makes sense — it would be annoying, for example, for preview popovers to appear immediately every time a link is hovered.

Author: Yup, like it. Done in the next commit. I've mostly used your suggestion, with a couple of tweaks.

#### HTML

We include a {{htmlelement("button")}}, a {{htmlelement("p")}}, and an {{htmlelement("input")}} of type `checkbox`. We specify the `<button>` as an interest invoker by giving it an `interestfor` attribute; its value is given as the `id` of the `<p>` element, therefore the paragraph is the invoker target. The paragraph is made into a popover by giving it a [`popover`](/en-US/docs/Web/HTML/Reference/Global_attributes/popover) attribute, causing it to be initially hidden.

```html live-sample___interest-invoker-delay
<button interestfor="mypopover">a button</button>
<p id="mypopover" popover>A hover toolip</p>
<form>
  <label for="apply-delay">Apply <code>interest-delay: 1s 2s</code></label>
  <input type="checkbox" id="apply-delay" />
</form>
```

#### CSS

In the CSS, we specify a rule with a `.delay` selector, which will apply an `interest-delay` value of `1s 2s` to any interest invoker the `delay` class is set on. We will set this on the `<button>` when the checkbox is checked using JavaScript.

```css live-sample___interest-invoker-delay
.delay {
  interest-delay: 1s 2s;
}
```

#### JavaScript

> In our script, we grab references to the `<button>` and the checkbox, then create an event listener that toggles setting and unsetting the `delay` class on the `<button>` when the checkbox value changes (is checked or unchecked).

Reviewer: ```suggestion
In our script, we get references to the `<button>` and the checkbox, then create an event listener that toggles the `delay` class on the `<button>` whenever the checkbox value changes (checked or unchecked).
```
